packages/client/src/components/ProductList/ProductList.component.js

```diff
@@ -3,11 +3,25 @@ import PropTypes from 'prop-types';
 import './ProductList.component.css';
 import ProductCard from '../ProductCard/ProductCard.component';
 
-export default function ProductList({ isLoading, products, filteredProducts }) {
+export default function ProductList({
+  isLoading,
+  products,
+  filteredProducts,
+  toggleFavorite,
+  favoriteProducts,
+}) {
   const ListOfProducts = filteredProducts.map((product) => {
     return (
       <div className="product">
-        <ProductCard title={product.name} price={product.price} />
+        <ProductCard
+          key={product.id}
+          title={product.name}
+          price={product.price}
+          id={product.id}
+          toggleFavorite={toggleFavorite}
+          isFavorite={favoriteProducts.some((x) => x.id === product.id)}
```

Comment:
For a small set of products this kind of code is probably fine, but if you have a lot of favorites this could become a problem since it will do this check logic on each render for each product for each favorite. 
This is why it's often better to figure out a way to join this kind of data as early as possible so it becomes one piece of data instead of something that needs to be "linked" on the fly every time it's needed.